j2cl-maven-plugin/src/main/java/com/vertispan/j2cl/mojo/TestMojo.java

```diff
@@ -312,7 +312,12 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             mainDep.setProject(main);
             testDeps.add(mainDep);
             test.setDependencies(testDeps);
-            test.setSourceRoots(new ArrayList<>(project.getTestCompileSourceRoots()));
+            test.setSourceRoots(project.getTestCompileSourceRoots().stream().filter(
+                  path -> 
+                        !(annotationProcessorMode.pluginShouldExcludeGeneratedAnnotationsDir()
+                            && path.endsWith(
+                                "generated-test-sources" + File.separator + "test-annotations")))
```

Comment:
could you put this instead into the AbstractBuildMojo? The field can stay private, and it will mean that compiling for production with scope=test (in the future, not supported today I believe) will still work.